### ISSUE TITLE
Keras / LSTM model saving and loading produces inconsistent results

### ISSUE THREAD

**Issue by navjitbrar** [keras-team/keras#11335]
I have seen several discussion on this topic in various forums. However, none of them seem to address the central issue I have. Here goes:

**Setup:**
- Working on a LSTM model. 
- It is a sentiment analysis application.
- I save / load the model per this FAQ (https://keras.io/getting-started/faq/#how-can-i-save-a-keras-model)
- It works if I do it from the *same context*. That is same program / .py file: Train / Save / Delete model / Load / Predict works.

**Problem**: 
The following does not work.
- Program 1: Train the model. Save the model.
- Program 2: Load the model. Do prediction.

The predictions are completely off when compared to a single context run.
- There is a similar issue - that had some discussion and then it was marked as stale. https://github.com/keras-team/keras/issues/4904
- Another work around seems here is here: https://github.com/keras-team/keras/issues/7632

**Question**:
- Do we have to store internal states in a some other pickle format - for LSTM cases?
- Is there another recommended way of what I am trying to attempt?

**Output of `pip freeze`**:
```
absl-py==0.4.1
astor==0.7.1
bleach==1.5.0
cycler==0.10.0
gast==0.2.0
grpcio==1.14.2
h5py==2.8.0
html5lib==0.9999999
Keras==2.2.2
Keras-Applications==1.0.4
Keras-Preprocessing==1.0.2
Markdown==2.6.11
matplotlib==2.1.1
numpy==1.14.5
pandas==0.22.0
protobuf==3.6.1
pyparsing==2.2.0
python-dateutil==2.6.1
pytz==2017.3
PyYAML==3.13
scikit-learn==0.19.1
scipy==1.1.0
six==1.11.0
sklearn==0.0
tensorboard==1.10.0
tensorflow==1.10.1
tensorflow-tensorboard==0.1.8
termcolor==1.1.0
Werkzeug==0.14.1
```

**Comment by navjitbrar** [comment]
It seems to be related to `return_state` option and my particular use. I am using the sequential API. Perhaps because I am using `binary_crossentropy` as a loss function, the following message was triggered after I set `return_state = True` : `All layers in a Sequential model should have a single output tensor. For multi-output layers, use the functional API.`

Exact code - with return_state removed:
```
embedding_vecor_length = 64
model = Sequential()
model.add(Embedding(ms.vocab_size, embedding_vecor_length, input_length=ms.max_review_length))
if ms.conv == 'y':
    model.add(Conv1D(filters=32, kernel_size=3, padding='same', activation='relu'))
    model.add(MaxPooling1D(pool_size=2))
model.add(LSTM(100, dropout=0.2, recurrent_dropout=0.2))
model.add(Dense(1, activation='sigmoid'))
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])
print(model.summary())
model.fit(X_train, y_train, epochs=ms.num_epoch, batch_size=64)
# Final evaluation of the model
scores = model.evaluate(X_test, y_test, verbose=0)
print("Accuracy: %.2f%%" % (scores[1]*100))

```
When I debug above in PyCharm, there are no values stored in `mode.layers[1].states`. If it were, perhaps it would make its way into the hd5 file - or - I perhaps I could glean it / pickle it and do the reverse on prediction run.

Going to move my application to the functional API and try it next.
